src/groups/orthogonal_group.jl

```diff
@@ -0,0 +1,283 @@
+"""
+    OrthogonalGroup{T}
+
+The orthogonal group ``$(_math(:O))(n)`` is the Lie group consisting of the [`MatrixMultiplicationGroupOperation`](@ref) on the
+manifold of rotations [`OrthogonalMatrices`](@extref `Manifolds.OrthogonalMatrices`).
+
+# Constructor
+    OrthogonalGroup(n; kwargs...)
+
+Generate  orthogonal group ``$(_math(:O))(n)``.
+All keyword arguments in `kwargs...` are passed on to [`OrthogonalMatrices`](@extref `Manifolds.OrthogonalMatrices`) as well.
+"""
+const OrthogonalGroup{T} = LieGroup{
+    ManifoldsBase.ℝ,MatrixMultiplicationGroupOperation,Manifolds.OrthogonalMatrices{T}
+}
+
+function OrthogonalGroup(n; kwargs...)
+    R = Manifolds.OrthogonalMatrices(n; kwargs...)
+    return OrthogonalGroup{typeof(R).parameters[1]}(R, MatrixMultiplicationGroupOperation())
+end
+
+#
+#
+# Generic special cases for O(n) and SO(n)
+
+_doc_exp_O2_id = """
+    exp(G::OrthogonalGroup{TypeParameter{Tuple{2}}}, ::Identity{MatrixMultiplicationGroupOperation}, X)
+    exp(G::SpecialOrthogonalGroup{TypeParameter{Tuple{2}}}, ::Identity{MatrixMultiplicationGroupOperation}, X)
+    exp!(G::OrthogonalGroup{TypeParameter{Tuple{2}}}, ::Identity{MatrixMultiplicationGroupOperation}, g, X)
+    exp!(G::SpecialOrthogonalGroup{TypeParameter{Tuple{2}}}, ::Identity{MatrixMultiplicationGroupOperation}, g, X)
```

Comment:
@mateuszbaran 

I do agree, that dispatching only on the type parameter here is a great idea. It however makes the docs (that I am carefully revising currently) a bit clumsy. Do you maybe have an idea what to best do with these 4 lines (analogously for log – and for both also the cases of (S)O(3) and 4)?
My concern here is mainly readability of the docs, the code is fine as is. It should convey that it only works for the the parameter case but still be readable

Reply:
Hm, it's somewhat problematic. I don't have any better idea for this docstring.

Reply:
I will think a bit about it. Maybe we can leave out the types in the signature and only mention them in the text. In the current length, sure we can read it, but for everyone else it is a bit too technical. Spo maybe just `(G, e, g, X)` is fine enough if all of them are explained in the next few lines, also that G has a type parameter to dispatch on.

Reply:
The simplified version is now without any types and they are only mentioned in the text – that also reduces the signatures to 2 instead of 4 – one could for sure still make it 4 and mention the Lie groups again for completeness. Current for is something like (colouring from REPL not complete here after copy ;))

```julia
  exp(G, e, X)
  exp!(G, e, g, X)

  Compute the Lie group exponential function on the OrthogonalGroup \mathrm{O}(4) or SpecialOrthogonalGroup
  \mathrm{SO}(4), where e is the Identity{MatrixMultiplicationGroupOperation} and G uses a TypeParameter (@extref
  `ManifoldsBase.TypeParameter`) for dispatch.

  Similar to the 3×3 case, an efficient computation is provided, adapted from GallierXu:2002 (@cite),
  AndricaRohan:2013 (@cite) with a few numerical stabilisations.
```